### PR TITLE
Feature/24 move flower

### PR DESCRIPTION
**Pull request by hromogen**
-- Move flower logic is added
-- Transactions  added as an instrument to make flower moves unbreakable
-- Some es-lint rules are adjusted according to the current project structure

### CODE REVIEW

**Review by dubbha — APPROVED** [review]

File: packages/server/src/db/db.service.test.ts [hunk 1]
```diff
@@ -269,29 +277,39 @@ describe('DBService', () => {
...
 
   it('should delete shelf invitation', async () => {
-    await new DBService().deleteShelfInvitation(mockShelfInvitationEntity.shelfId);
-    expect(mockShelfInvitationRepository.delete).toBeCalledWith(mockShelfInvitationEntity.shelfId);
+    await new DBService().deleteShelfInvitation(
+      mockShelfInvitationEntity.shelfId
+    );
+    expect(mockShelfInvitationRepository.delete).toBeCalledWith(
+      mockShelfInvitationEntity.shelfId
+    );
```
Comment (dubbha) [inline packages/server/src/db/db.service.test.ts]: I still feel like you have prettier running even though we have it removed. Too many changes that should not be enforced by our current eslint rules make PRs really hard to comprehend.

File: packages/server/src/db/db.service.ts [hunk 2]
```diff
@@ -219,7 +228,7 @@ export default class DBService {
       where: { userId },
       relations: ['shelf'],
     });
-    return userToShelves.map(u2s => u2s.shelf);
+    return userToShelves.map((u2s) => u2s.shelf);
```
Comment (dubbha) [inline packages/server/src/db/db.service.ts]: There's no eslint rule enforcing this. Why does it still get changed?

File: packages/server/src/db/db.service.ts [hunk 3]
```diff
@@ -302,40 +311,107 @@ export default class DBService {
...
+    });
+    return userToShelves.map((u2s) => u2s.user);
   }
 
-  async saveAction(userId: number, flowerId: number, action: Actions, timestamp: number) {
+  async saveAction(
+    userId: number,
+    flowerId: number,
+    action: Actions,
+    timestamp: number
```
Comment (dubbha) [inline packages/server/src/db/db.service.ts]: There's no eslint rule enforcing this. Why does it still get changed? Are you changing these manually?

File: packages/server/src/shelf/shelf-move-flower.dto.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { IsNumber } from 'class-validator';
+
+export default class ShelfMoveFlowerDto {
+  @IsNumber()
+  flowerId: number;
+  @IsNumber()
+  shelfId: number;
+  @IsNumber()
+  targetShelfId: number;
```
Comment (dubbha) [inline packages/server/src/shelf/shelf-move-flower.dto.ts:11]: Maybe ad blank lines between them

File: packages/server/src/shelf/shelf.controller.ts [hunk 5]
```diff
@@ -129,15 +136,12 @@ export default class ShelfController extends BaseController {
     next: NextFunction
   ): Promise<void> => {
     try {
-      await this.shelfService.inviteUser(
-        request.body,
-        request.headers.origin,
-      );
+      await this.shelfService.inviteUser(request.body, request.headers.origin);
```
Comment (dubbha) [inline packages/server/src/shelf/shelf.controller.ts]: There's no eslint rule enforcing this. Why does it still get changed?

File: packages/server/src/shelf/shelf.controller.ts [hunk 6]
```diff
@@ -146,13 +150,13 @@ export default class ShelfController extends BaseController {
   ): Promise<void> => {
     try {
       await this.shelfService.acceptInvitation(
-        request.body.shelfInvitationToken,
+        request.body.shelfInvitationToken
```
Comment (dubbha) [inline packages/server/src/shelf/shelf.controller.ts]: There's no eslint rule enforcing this. Why does it still get changed?

File: packages/server/src/shelf/shelf.controller.ts [hunk 7]
```diff
@@ -176,13 +180,13 @@ export default class ShelfController extends BaseController {
     try {
       await this.shelfService.addShelf(
         request.body,
-        request.headers.authorization,
+        request.headers.authorization
```
Comment (dubbha) [inline packages/server/src/shelf/shelf.controller.ts]: There's no eslint rule enforcing this. Why does it still get changed?

**Review by 0to1to0to1 — APPROVED** [review]

**Review by dartwlad — APPROVED** [review]

**Pull request merged**